### DISCUSSION TITLE
MBXP 1.29

### DISCUSSION THREAD

**Discussion by tlefko** [tlefko/Matebook-X-Pro-2018-Monterey#47] [Announcements]
***Major Update***

- Revised the hotplug system thanks to feedback from @yonathanarya
- Update CPU PM
- Update Battery Ticker
- Update Background Refresh
- This enables a complete working Type-C TBT hotplug with PCI-E and USB Bridge Capabilities
- Native Video-out and docking support
- Increased reliability and stability for hotplug and HDMI-Out.
- Native External GPU Support via -raddvi boot args (if needing to use DVI for AMD)

- **This Repository Now Supports MBXP2019**

**Full Changelog**: https://github.com/tlefko/Matebook-X-Pro-2018-Ventura/compare/venturamajor2...gigamajor

**Full Changelog**: https://github.com/tlefko/Huawei-Matebook-X-Pro-Ventura/compare/gigamajor...gigamajorss

<hr /><em>This discussion was created from the release <a href='https://github.com/tlefko/Huawei-Matebook-X-Pro-Ventura/releases/tag/'>MBXP 1.29</a>.</em>

**Comment by Paxolo** [comment]
Hi, I hope the exam season is going well :)
I tried this latest EFI on Ventura 13.2.1 and unfortunately built-in peripherals are still broken for me.
I saw you didn't mention any fixes in the changelog, though, so I imagine this release was focused on getting Thunderbolt working instead, which unfortunately is kind of hard for me to test :(
For the time being, I'll just downgrade to the previous version

**Reply by tlefko** [reply]
@Paxolo yeah, for your machine specifically I am little stumped haha. if you send me your nworking downgraded EFI i will patch in the thunderbolt changes.

**Reply by Paxolo** [reply]
I didn't know it was a machine-specific issue, that's sad 😢 
Do you think making my own ACPI configuration could fix it?
If so, would you mind sharing a guide for me to do it? Dortania's is very complete but I find it a bit unclear in a way

**Comment by Leon5921** [comment]
Hi I'm running a Matebook X pro 2020 on ventura. I was wondering if the thunderbolt hotplug system would be applicable to my system.

**Reply by yonathanarya** [reply]
It is designed to work with Alpine Ridge Controller. If your MBXP2020 using the same JHL6XXX series thunderbolt controller, then it might work

**Comment by QIYI0320** [comment]
I used the boss's efi on macos13.3. I couldn't boot successfully. At the moment when I reached the Apple icon and the progress bar flashed, my computer screen restarted with a black screen. After that, the blue screen of windows could not enter. Do you know the reason?
